cluster/join.go

```diff
@@ -50,11 +50,20 @@ func Join(joinAddr []string, id, addr string, voter bool, meta map[string]string
 	return "", ErrJoinFailed
 }
 
-func join(joinAddr, id, addr string, voter bool, meta map[string]string, tlsConfig *tls.Config, logger *log.Logger) (string, error) {
+func join(sourceIp string, joinAddr, id, addr string, voter bool, meta map[string]string, tlsConfig *tls.Config, logger *log.Logger) (string, error) {
```

Comment:
Go variable style states this should `sourceIP` -- and `srcIP` would be better (Go prefers short names). Also, `string` is not required as all types following are the same until `voter`.